src/userSync.js

```diff
@@ -0,0 +1,175 @@
+import * as utils from 'src/utils';
+
+export function newUserSync() {
+  // Set user sync config default values which can be overridden by the publisher
+  const userSyncDefaultConfig = {
+    syncEnabled: true,
+    pixelEnabled: true,
+    syncsPerBidder: 5
+  }
+
+  let publicApi = {};
+  // A queue of user syncs for each adapter
+  // Let getDefaultQueue() set the defaults
+  let queue = getDefaultQueue();
+
+  // Since user syncs require cookie access we want to prevent sending syncs if cookies are not supported
+  let cookiesAreSupported = !utils.isSafariBrowser() && utils.cookiesAreEnabled();
+  // Whether or not user syncs have been trigger on this page load
+  let hasFired = false;
+  // How many bids for each adapter
+  let numAdapterBids = {};
+
+  // Merge the defaults with the user-defined config
+  let userSyncConfig = Object.assign($$PREBID_GLOBAL$$.userSync || {},
+    userSyncDefaultConfig);
```

Comment:
That's fair. I thinking passing it in is fine. I think it would make testing easier too.